### PR TITLE
[HPA/e2e] Add nil checks to behavior validation and reduce stabilization windows 

### PR DESCRIPTION
**Pull request by moh-osman3**
Some fixes to the e2e tests to help resolve helm test failure from https://github.com/open-telemetry/opentelemetry-operator/issues/1574

### CODE REVIEW

File: tests/e2e/autoscale/02-install.yaml [hunk 1]
```diff
@@ -11,13 +11,11 @@ spec:
...
     targetMemoryUtilization: 70
-    # Without this behavior the HPA will default to a scaledown stabilization
-    # window of 300 seconds. Tests should fail if this update is not successful.
     behavior:
       scaleDown:
-        stabilizationWindowSeconds: 15
+        stabilizationWindowSeconds: 1
       scaleUp:
-        stabilizationWindowSeconds: 25
+        stabilizationWindowSeconds: 1
```
Comment (pavolloffay) [inline tests/e2e/autoscale/02-install.yaml:18]: what is the motivation of setting both of these to 1s?

Comment (moh-osman3) [reply tests/e2e/autoscale/02-install.yaml]: No longer changing scaleUp, but I am trying to remove any downtime in scale down. The helm repo seems to be using `kubernetes v1.23` and for some reason the scaleDown option is being ignored (and the default of 300s is used).  I now also added a timeout to at least unblock the helm tests from running. Do you have any thoughts on best way forward for https://github.com/open-telemetry/opentelemetry-operator/issues/1574?

Comment (iblancasa) [reply tests/e2e/autoscale/02-install.yaml]: @moh-osman3 [they are using 1.25](https://github.com/open-telemetry/opentelemetry-helm-charts/actions/runs/4424164450/jobs/7757660240#step:3:117)

File: tests/e2e/autoscale/05-assert.yaml [hunk 2]
```diff
@@ -1,3 +1,7 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+timeout: 360
```
Comment (iblancasa) [inline tests/e2e/autoscale/05-assert.yaml:3]: Please, don't increase the timeout

**Comment by pavolloffay** [comment]
Replaces by #1582

**Pull request closed without merging**